### DISCUSSION TITLE
Support for JavaScript and Typescript

### DISCUSSION THREAD

**Discussion by FadySalama** [ls1intum/Artemis#4658] [Q&A] [answered]
Hello,

I have a question regarding support for JavaScript and Typescript exercises and automated testing.  Is it planned to add such a feature? And if not, is it possible to automatically test the submissions of students using docker images? What steps should I take to make it possible? I would appreciate the help

Thank you!

**Comment by krusche** [comment]
You can already add JavaScript and Typescript exercises manually. We are happy to develop a template together with you, so that the creation is easier. Yes, programming exercises are typically tested in Docker container, so you would need to specify a (custom) container in the programming exercise, e.g. with Node.js

**Reply by FadySalama** [reply]
Would it be possible to show me the steps needed to set up the testing? I am having a problem understanding how to do the testing in the case of a JavaScript/Typescript repo.

**Reply by Mtze** [reply]
You can find some example docker images / Dockerfiles here: 
- https://github.com/ls1intum/artemis-maven-docker
- https://github.com/ls1intum/artemis-c-docker
- https://github.com/ls1intum/artemis-python-docker
- https://github.com/ls1intum/artemis-ocaml-docker

Make sure that the software tests you define can be executed in the container.

**Reply by FadySalama** [reply]
How can I include that dockerfile in the BASE and SOLUTION Build Plan? I try for example by adding the docker image under the plans docker configuration (see image below). However, all jobs that I try to execute don't work as intended. Mainly, I try having a specific version of Node.js running but I always get a much older version in when I test the version. I just need help with how you configure build plans to work and what does each build plan do and how to use a docker image with said build plan

<img width="765" alt="image" src="https://user-images.githubusercontent.com/24440113/164287619-0b90f577-6e48-4635-8a6d-26e48c915f6e.png">

**Comment by MaximilianAnzinger** [comment, marked as answer]
Artemis now supports JS & TS. Tinkering with custom docker containers is not necessary anymore 👍
